### PR TITLE
Update Code Walktrough to match current template

### PR DESCRIPTION
**Pull request by vigejolla**
Also add a paragraph about X-SailJail with a link to sandboxing documentation.

### CODE REVIEW

**Review by jpwalden — APPROVED** [review]
as far as I can see this looks good to me.

File: Develop/Apps/Code_Walkthrough/README.md [hunk 1]
```diff
@@ -52,23 +52,26 @@ int main(int argc, char *argv[])
...
 ApplicationWindow
 {
     initialPage: Component { FirstPage { } }
     cover: Qt.resolvedUrl("cover/CoverPage.qml")
+    allowedOrientations: defaultAllowedOrientations
 }
 ```
 
-ApplicationWindow is the top-level type of all Sailfish Silica applications. The initialPage property specifies the first page to display when the application is opened. The cover property sets the active cover to be displayed when the application is pushed to the background.
+`ApplicationWindow` is the top-level type of all Sailfish Silica applications. The `initialPage` property specifies the first page to display when the application is opened. The `cover` property sets the active cover to be displayed when the application is pushed to the background.
```
Comment (rainemak) [inline Develop/Apps/Code_Walkthrough/README.md:72]: I'd first introduce here 3 concepts: [ApplicationWindow](https://sailfishos.org/develop/docs/silica/qml-sailfishsilica-sailfish-silica-applicationwindow.html/), Page, and PageStack. This contains a bit repeated content from ApplicationWindow doc but would work as introduction. Maybe something like this:

```suggestion
[ApplicationWindow](https://sailfishos.org/develop/docs/silica/qml-sailfishsilica-sailfish-silica-applicationwindow.html/) is the top-level type of all Sailfish Silica applications. Each application window contains a single page stack, accessible through the [pageStack](https://sailfishos.org/develop/docs/silica/qml-sailfishsilica-sailfish-silica-applicationwindow.html/#pageStack-prop) property, that determines the content to be displayed by the application. The stack is made up of [Page](https://sailfishos.org/develop/docs/silica/qml-sailfishsilica-sailfish-silica-page.html) objects.

The [initialPage](https://sailfishos.org/develop/docs/silica/qml-sailfishsilica-sailfish-silica-applicationwindow.html/#initialPage-prop) property specifies the first page to display when the application is opened. The [cover](https://sailfishos.org/develop/docs/silica/qml-sailfishsilica-sailfish-silica-applicationwindow.html/#cover-prop) property sets the active cover to be displayed when the application is pushed to the background.
```

File: Develop/Apps/Code_Walkthrough/README.md [hunk 2]
```diff
@@ -78,72 +81,82 @@ ApplicationWindow is the top-level type of all Sailfish Silica applications. The
 ```qml
 Page {
     id: page
+    allowedOrientations: Orientation.All
 ```
+Here, we first create a `Page` object which is simply a container for the contents of the page. With `allowedOrientations` we could restrict the allowed orientations of the page. Effectively the set of allowed orientations is an intersection of the allowed orientations in the `ApplicationWindow` and the `Page`.
+
+To enable a pull-down menu, we need to create a flickable item which places its children on a surface that can be pulled and flicked. We do this using the `SilicaFlickable` component. Sailfish Silica provides the types `SilicaFlickable`, `SilicaListView`, and `SilicaGridView` which are styled versions of the Qt Quick flickable, list view, and grid view types.
```
Comment (rainemak) [inline Develop/Apps/Code_Walkthrough/README.md:88]: To ease navigation we should have clickable api ref links always when referring to Sailfish/Silica components. I this article that'd be: ApplicationWindow, Page, SilicaFlickable, SilicaListView, SilicaGridView , MenuItem, PullDownMenu, PushUpMenu, PageHeader, Label, Theme, CoverBackground, CoverActionList, CoverAction, and TextField (given that I checked them right).

File: Develop/Apps/Code_Walkthrough/README.md [hunk 3]
```diff
@@ -184,27 +197,36 @@ CoverActionList {
...
 
 ### Other Files and Directories in This Project
 
 You will notice that in addition to the .qml and .cpp files, there are a couple of other files that we have not touched so far. Let’s quickly go through them before you start exploring the SDK.
 
-`myfirstapp.desktop` is a standard Linux desktop configuration file that describes how a particular program is to be launched, how it appears in menus, etc. This file specifies for example the name and icon of your applcation as they appear in the launcher. More information on the desktop entry specification can be found at standards.freedesktop.org.
+`myfirstapp.desktop` is a standard Linux desktop configuration file that describes how a particular program is to be launched, how it appears in menus, etc. This file specifies for example the name and icon of your applcation as they appear in the launcher. More information on the `Desktop Entry` specification can be found at [https://standards.freedesktop.org](Freedesktop.org Specifications) page.
```
Comment (rainemak) [inline Develop/Apps/Code_Walkthrough/README.md:216]: Broken freedesktop link:
```suggestion
`myfirstapp.desktop` is a standard Linux desktop configuration file that describes how a particular program is to be launched, how it appears in menus, etc. This file specifies for example the name and icon of your applcation as they appear in the launcher. More information on the `Desktop Entry` specification can be found at [Freedesktop.org Specifications](https://standards.freedesktop.org) page.
```

**Review by rainemak — COMMENTED** [review]
Added few suggestions. I think most beneficial for readers when going through this code walktrough would be links to API docs. That would nicely tie this to the docs.

In this article I'd add link to: ApplicationWindow, Page, SilicaFlickable, SilicaListView, SilicaGridView , MenuItem, PullDownMenu, PushUpMenu, PageHeader, Label, Theme, CoverBackground, CoverActionList, CoverAction, and TextField (given that I checked them right).

**Comment by vigejolla** [comment]
> Added few suggestions. 

These should all be done now

**Review by rainemak — APPROVED** [review]

**Pull request merged**